### PR TITLE
LF-4339: Create product form validation

### PR DESCRIPTION
**Pull request by zagran**
**Description**

I added validation of trimmed value length for the required field Supplier. This will fix the issue on this form only, but theoretically, we can add this check on all `required` validation options.

Jira link: https://lite-farm.atlassian.net/browse/LF-4339 

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx [hunk 1]
```diff
@@ -366,6 +366,12 @@ const ProductDetails = ({
             hookFormRegister={register(SUPPLIER, {
               required: interested,
               maxLength: hookFormMaxCharsValidation(255),
+              validate: (value) => {
+                if (value?.trim().length === 0) {
+                  return t('COMMON_ERRORS.UNIT.REQUIRED');
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx:371]: So it is actually only conditionally required if certifications `interested` is `true`.

So currently this is also validating in the `optional` case.

**Review by Duncan-Brain — CHANGES_REQUESTED** [review]
Just looking at the video in the Jira ticket it looks like there is a special case where you can click `Create " "` in the creatable select when there are no other pre-existing products (I had to try to create a new farm to see this) . And I can still do that and receive an error on this branch. If it feels out of scope feel free to duplicate the Jira ticket and rename it focusing on supplier only for this PR.

But main issue I see currently is that the optional field shouldn't validate -- only when it is required.

File: packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx (hunk 1 above)
Comment (zagran) [reply packages/webapp/src/components/Task/AddSoilAmendmentProducts/ProductCard/ProductDetails.tsx:371]: @Duncan-Brain Fixed, thanks for the feedback

**Comment by zagran** [comment]
> Just looking at the video in the Jira ticket it looks like there is a special case where you can click `Create " "` in the creatable select when there are no other pre-existing products (I had to try to create a new farm to see this) . And I can still do that and receive an error on this branch. If it feels out of scope feel free to duplicate the Jira ticket and rename it focusing on supplier only for this PR.
> 
> But main issue I see currently is that the optional field shouldn't validate -- only when it is required.

Oh, you're absolutely right, thanks for pointing this out to me (especially how to test it via creating a new farm)
I think it works as expected now, I tested it on a new farm.

**Review by Duncan-Brain — APPROVED** [review]
Thanks for making the changes and fixing the creatable select!

**Review by kathyavini — APPROVED** [review]
Sorry for the slow review 🙇 

Thank you for fixing and for looking into all those edge cases too!

**Pull request merged**